packages/opentelemetry-exporter-prometheus/src/PrometheusLabelsBatcher.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  MetricRecord,
+  MetricDescriptor,
+  AggregatorKind,
+} from '@opentelemetry/metrics';
+import { PrometheusCheckpoint } from './types';
+
+interface BatcherCheckpoint {
+  descriptor: MetricDescriptor;
+  aggregatorKind: AggregatorKind;
+  records: Map<string, MetricRecord>;
+}
+
+export class PrometheusLabelsBatcher {
+  private _batchMap = new Map<string, BatcherCheckpoint>();
+
+  get hasMetric() {
+    return this._batchMap.size > 0;
+  }
+
+  process(record: MetricRecord) {
+    const name = record.descriptor.name;
+    let item = this._batchMap.get(name);
+    if (item == null) {
```

Comment:
Is there a particular reason for which you didnt use a strict comparison here ? I believe this should be `item === undefined`

Reply:
yes, == can compare loosely with null and undefined. It's same as item === null && item === undefined.

Reply:
I actually prefer not to treat null and undefined differently unless there is a good reason for it. I am fine with `==` comparisons but I would like @open-telemetry/javascript-maintainers opinions here.

Reply:
I mean at no point in this code the `item` should actually be equal `null` (as either its undefined or it contain a value, i find it misleading for new contributors who might not be familiar with the loosy comparison